packages/jest-mock/src/index.ts

```diff
@@ -20,87 +20,101 @@ export type MockFunctionMetadataType =
 export type MockFunctionMetadata<
   T,
   Y extends Array<unknown>,
-  Type = MockFunctionMetadataType,
+  MetadataType = MockFunctionMetadataType,
 > = {
   ref?: number;
   members?: Record<string, MockFunctionMetadata<T, Y>>;
   mockImpl?: (...args: Y) => T;
   name?: string;
   refID?: number;
-  type?: Type;
+  type?: MetadataType;
   value?: T;
   length?: number;
 };
 
-export type MockableFunction = (...args: Array<any>) => any;
-export type MethodKeysOf<T> = {
-  [K in keyof T]: T[K] extends MockableFunction ? K : never;
+export type ConstructorLike = {new (...args: Array<any>): any};
```

Comment:
`Constructable`?

Reply:
from TS that is, not a renaming suggestion

Reply:
Hm.. Does not work for me. Should it be somehow imported or simply present globally as say `Promise` or `Array`? Currently `Constructable` is defined here:

https://github.com/facebook/jest/blob/c2cd7e6d415517b5c9f8399bbf6b43c7c1dec663/packages/jest-mock/src/index.ts#L95-L97

Reply:
Ah, I was fooled by an auto-import 🙈 

New suggestion then - using `type-fest`: https://github.com/sindresorhus/type-fest/blob/cbd7ec510bd136ac045bbc74e391ee686b8a9a2f/source/basic.d.ts#L13

Reply:
Hm.. Not sure. It isn’t exactly the same thing (I mean, not doing same job) and it feels too much to pull in a dependency for just one line of code. I prefer writing type tests ;D That’s also overhead, of course.

Reply:
if it's only that single utility then it's not worth it 👍